### ISSUE TITLE
Add an internal renderer for the background

### ISSUE THREAD

**Issue by CrimsonFork** [stanipintjuk/Silverfish#48]
On my phone (Neffos C5L) I have the problem that the image resizer doesn't work properly. The higher the resolution is the more ugly it's scaled.

I don't know if the API of Android allows to read the background image but if yes it would be cool if it would be resized by Silverfish and used as an internal background, if you know what I mean.

**Comment by Lonami** [comment]
I assume that by "internal background" you mean "wallpaper"? It's the one used on the phone, there's no further modification by the launcher itself.

What application are you using to set the wallpaper? Have you tried other options?

**Comment by CrimsonFork** [comment]
I actually meant "**internal renderer** for the background".

There's no difference if I use the builtin gallery app or Simple Gallery, the result doesn't change. Also if I set a backgroud picture inside am app it looks just as ugly as in the "normal" background.

**Comment by Lonami** [comment]
> I don't know if the API of Android allows to read the background image

Yes, [it does](http://stackoverflow.com/a/9939358/4759433) (and is indeed as simple as `WallpaperManager.getInstance(this).getDrawable()`). But this would introduce another problem: live wallpapers *would not work*.

**Comment by CrimsonFork** [comment]
Well that's no problem (for me at least). I have no (graphical) problems with wallpapers so if it's not complicated to let Silverfish easily detect if it's a picture or not (it should, but I'm not a programmer, I don't know) this would solve the ptoblem.

**Comment by Lonami** [comment]
> Well that's no problem (for me at least).

That's the problem. Some people might like and be using live wallpapers, and if we retrieved the wallpaper *picture* (this is, something static), that would not work. Also, some live wallpapers have user interaction too (they're not simple "movies").

It is possible to detect whether it's a live wallpaper or not, but implementing this would be a bit odd since the whole application theme needs to change to either use or not the wallpaper "automatically" I think.

**Comment by CrimsonFork** [comment]
Well then maybe an option in the settings menu would help as it looks like it is going to exist anyway (the menu)?
